Add the option to hide both the task and command lines in the task output

hugocardante opened the pull request:
The goal is to be able to hide these lines from a task output:

```sh
⏵ Task `...` finished successfully
⏵ Command: ...
```

Here is a video that describes the functionality:
https://github.com/user-attachments/assets/4f17e530-ca53-46df-bbbd-84bcc618ac86
(feel free to watch it in 2x as it's almost 2 minutes long)

Release Notes:

- Added a way to hide both the task and command lines in the task output

Commit:
Added the option to hide both the task and command lines in the task output

omennia commented:
@cla-bot check

notpeter commented on crates/project/src/terminals.rs:
```diff
@@ -174,6 +174,8 @@ impl Project {
                     command_label: spawn_task.command_label,
                     hide: spawn_task.hide,
                     status: TaskStatus::Running,
+                    hide_task: spawn_task.hide_task,
```
I don't mean to bikeshed, but I think it's potentially confusing for `hide` to control the task terminal tab behavior and `hide_*` to suppress certain output inside the terminal. E.g:

```
  {
    "label": "Echo Echo",
    "command": "echo ECHO; sleep 1; echo echo; sleep 1",
    "hide": "never",
    "hide_task": true,
    "hide_command": true
  },
```

Looks like vscode has an entire `presentation` key (your `hide_command` corresponds to `presentation.echo = false`):
```json
//defaults
"presentation": {
    "echo": true,
    "reveal": "always",
    "focus": false,
    "panel": "shared",
    "showReuseMessage": true,
    "clear": false
},
```

Perhaps substitue:
- `show_command` for `hide_command` (default true)
- `show_summary` for `hide_task` (default true)

We should also document these: [tasks.md](https://github.com/zed-industries/zed/blob/main/docs/src/tasks.md?plain=1)

notpeter replied:
Otherwise this is awesome.
I didn't realize I wanted clean task output until I tried it. 👍

omennia replied:
Hello, thank you for the feedback.
After seeing your comment I realize how much sense it makes.

Also, learned a new expression - bikeshed (not native English speaker)

SomeoneToIgnore commented on crates/terminal/src/terminal.rs:
```diff
@@ -1760,11 +1762,24 @@ impl Terminal {
         // when Zed task finishes and no more output is made.
         // After the task summary is output once, no more text is appended to the terminal.
-        unsafe { append_text_to_term(&mut self.term.lock(), &[&task_line, &command_line]) };
+        // Only append text if there are lines to show
```
Passing by and noticing that this comment is useless (as `!..is_empty()` code shows it much better) + the SAFETY comment is not in the right position now (needs to be moved inside the `if` block).

`Decide whether to display the task and/or command lines based on user preferences.` is also useless and not needed.

omennia replied:
Thanks for the feedback! Appreciate it

Commit:
Rename visibility flags and improve documentation

- Rename hide_task to show_task
- Rename hide_command to show_command
- Fix comments in terminal.rs
- Update task documentation in tasks.md

Commit:
corrected typo

notpeter committed:
Rename show_task to show_summary

notpeter commented:
I renamed `show_task` to `show_summary` to avoid any confusion, but otherwise this is great!

It'll be available in Zed v0.164.x which will be in Preview Wednesday 2024-11-27 and in Stable the following week! 

Thanks and congrats on your first contribution to Zed!
